src/core/common/message.cpp

```diff
@@ -294,12 +294,34 @@ ThreadError Message::Prepend(const void *aBuf, uint16_t aLength)
     mInfo.mLength += aLength;
     SetOffset(GetOffset() + aLength);
 
-    Write(0, aLength, aBuf);
+    if (aBuf != NULL)
+    {
+        Write(0, aLength, aBuf);
+    }
 
 exit:
     return error;
 }
 
+ThreadError Message::RemoveHeader(uint16_t aLength)
```

Comment:
Side note: On my local branch i have already working IP-in-IP encapsulation, and i read IPv6 Header more like `aMessage.Read(aMessage.GetOffset(), ...` than `aMessage.Read(0, ...`. However i will use the ocasion to ask, if for IP-in-IP encapsulation you would rather increase the Offset of the message to parse encapsulated IPv6 Header or use your new RemoveHeader method and just remove encapsulating header with ff03::fc address? Now i see a good reason to use RemoveHeader if some application make a Clone() of the message, but we don't want  encapsulating header to be included.